Ginger/GingerCoreCommon/Repository/SolutionRepository.cs

```diff
@@ -345,14 +355,38 @@ private SolutionRepositoryItemInfoBase GetSolutionRepositoryItemInfo(Type type)
 
             return SRII;
         }
-
+     
         private SolutionRepositoryItemInfo<T> GetSolutionRepositoryItemInfo<T>()
         {
             SolutionRepositoryItemInfoBase SRIIBase = GetSolutionRepositoryItemInfo(typeof(T));
             SolutionRepositoryItemInfo<T> SRII = (SolutionRepositoryItemInfo<T>)SRIIBase;
             return SRII;
         }
 
+ 
+        public RepositoryItemBase GetRepositoryItemByPath(string path)
+        {           
+            RepositoryItemBase repoItem = null;
+            Parallel.ForEach(mSolutionRootFolders, folder =>
+            {
+                if (Path.GetDirectoryName(path).Contains(folder.FolderFullPath))                
+                {                    
+                    ObservableList<RepositoryItemBase> repoItemList = folder.GetFolderRepositoryItems();
+                    repoItem = repoItemList.Where(x => Path.GetFullPath(x.FileName) == Path.GetFullPath(path)).FirstOrDefault();
+                    if (repoItem == null)
+                    {
+                        RepositoryFolderBase subfolder = folder.GetSubFolderByName(folder.FolderRelativePath + "\\" + Path.GetFileName(Path.GetDirectoryName(path)), true);
+                        if (subfolder.FolderFullPath.Contains(Path.GetDirectoryName(path)))
+                        {
+                            ObservableList<RepositoryItemBase> test2 = subfolder.GetFolderRepositoryItems();
+                            repoItem = test2.Where(x => Path.GetFullPath(x.FileName) == Path.GetFullPath(path)).FirstOrDefault();
+                        }
+                    }
+                }
+            });
```

Comment:
Please split this function to GetRepositoryItemByPath(string itemFilePath) and GetRepositoryFolderByPath(string folderPath)